polaris-react/src/components/ActionMenu/components/MenuGroup/MenuGroup.tsx

```diff
@@ -42,6 +42,14 @@ export function MenuGroup({
     onClose(title);
   }, [onClose, title]);
 
+  const handleCloseActionList = useCallback(
+    (id?: string) => {
+      const hasRef = actions.find((action) => action.id === id)?.ref;
```

Comment:
```suggestion
      const hasRef = actions.find((action) => action.id && action.id === id)?.ref; 
```[
](https://github.com/Shopify/web/pull/87281/files#r1146974853)
Let's check for a truthy `action.id` to make sure we don't evaluate to true when both are `undefined`